bundles/org.openhab.core.model.script.runtime/src/org/openhab/core/model/script/runtime/internal/engine/DSLScriptEngine.java

```diff
@@ -87,10 +87,9 @@ public Object eval(String script) throws ScriptException {
             if (script.stripLeading().startsWith(DSLScriptContextProvider.CONTEXT_IDENTIFIER)) {
                 String contextString = script.stripLeading().substring(
                         DSLScriptContextProvider.CONTEXT_IDENTIFIER.length(), script.stripLeading().indexOf('\n'));
-                String[] segments = contextString.split("-");
-                if (segments.length == 2) {
-                    modelName = segments[0];
-                    String ruleIndex = segments[1];
+                if (contextString.contains("-")) {
+                    modelName = contextString.substring(0, contextString.lastIndexOf('-'));
+                    String ruleIndex = contextString.substring(contextString.lastIndexOf('-') + 1);
```

Comment:
We can store `contextString.lastIndexOf('-')` in a local variable and reuse it:

```suggestion
                int index = contextString.lastIndexOf('-');
                if (index != -1) {
                    modelName = contextString.substring(0, index);
                    String ruleIndex = contextString.substring(index + 1);
```

Reply:
addressed